packages/launcher/src/renderer/launcher/wallets/WalletsView.js

```diff
@@ -84,10 +94,64 @@ const addWalletMutation = graphql`
   }
 `
 
+const TOKEN_ADDRESS = {
+  testnet: '0xa46f1563984209fe47f8236f8b01a03f03f957e4',
+  mainnet: '0xdf2c7238198ad8b389666574f2d8bc411a4b7428',
+}
+
+const newEthRequest = (id, method, params) => {
+  return {
+    id: id,
+    jsonrpc: '2.0',
+    method,
+    params,
+  }
+}
+
 class WalletsView extends Component<Props, State> {
   static contextType = EnvironmentContext
 
-  state = {}
+  state = {
+    balances: {},
+  }
+
+  componentDidMount() {
+    this.getBalances()
+  }
+
+  async getBalances() {
```

Comment:
Could we put this logic in the daemon rather than the launcher please?
I think the flow could be much simpler if it was handled on the daemon side, and getting these values might be needed at some point by internal logic.